pulp_rpm/app/kickstart/treeinfo.py

```diff
@@ -63,9 +94,13 @@ def deserialize(self, parser):
         try:
             super().deserialize(parser)
         except Exception:
+            self.sections_from_general(parser)
+            deserialize_order = "header release tree variants checksums images stage2 media".split()
             sections = parser._sections.keys()
 
-            for section in sections:
+            for section in deserialize_order:
```

Comment:
ProductMD follows a specific order to deserialize:
https://github.com/release-engineering/productmd/blob/master/productmd/treeinfo.py#L120

Reply:
it would be helpful to have it in the comments so anyone who looks later at the code understood why there is some special handling present.